@BurningEnlightenment ,Thanks for your pr, when I tested the features locally by run command "./vcpkg install stduuid[*]:x64-windows" but failed:
````
Stored binary cache: C:\Users\test\AppData\Local\vcpkg\archives\3a\3aeb5f916ec5358eb5a509ef887e2f36dd9128b010edf877c7bedda9dc561293.zip
The following files are already installed in F:/Feature-test/stduuid/vcpkg/installed/x86-windows and are in conflict with stduuid:x86-windows

Installed by ms-gsl:x86-windows
    include/gsl/gsl
    include/gsl/gsl_algorithm
    include/gsl/gsl_assert
    include/gsl/gsl_byte
    include/gsl/gsl_util
    include/gsl/pointers
    include/gsl/span
    include/gsl/string_span

Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
````
when I run command "./vcpkg install stduuid[*]:x64-linux" also failed:
````
OT_DIR=/home/test/test/vcpkg -D_VCPKG_INSTALLED_DIR=/home/test/test/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF
    Working Directory: /home/test/test/vcpkg/buildtrees/stduuid/x64-linux-rel
    Error code: 1
    See logs for more information:
      /home/test/test/vcpkg/buildtrees/stduuid/config-x64-linux-rel-out.log
      /home/test/test/vcpkg/buildtrees/stduuid/config-x64-linux-rel-err.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:355 (vcpkg_execute_required_process)
  ports/stduuid/portfile.cmake:19 (vcpkg_cmake_configure)
  scripts/ports.cmake:146 (include)
````
/home/test/test/vcpkg/buildtrees/stduuid/config-x64-linux-rel-err.log
````
CMake Error at cmake/FindLibuuid.cmake:11 (message):
  Could NOT find libuuid
Call Stack (most recent call first):
  /home/test/test/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package)
  CMakeLists.txt:25 (find_package)
````

--- a/ports/stduuid/fix-gsl-polyfill.patch
+++ b/ports/stduuid/fix-gsl-polyfill.patch
@@ -1,0 +1,37 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f45a7dc..039ee53 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -37,11 +37,11 @@ if (UUID_TIME_GENERATOR)
+ endif()
+ 
+ # Using span from std
+-if (NOT UUID_USING_CXX20_SPAN)
+-    target_include_directories(${PROJECT_NAME} INTERFACE
+-            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/gsl>
+-            $<INSTALL_INTERFACE:include/gsl>)
+-    install(DIRECTORY gsl DESTINATION include)
++if (UUID_USING_CXX20_SPAN)
++    target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_20)
++else()
++    find_package(Microsoft.GSL CONFIG REQUIRED)
++    target_link_libraries(${PROJECT_NAME} INTERFACE Microsoft.GSL::GSL)
+ endif ()
+ 
+ # Install step and imported target
+diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
+index 7217b72..fb981d2 100644
+--- a/cmake/Config.cmake.in
++++ b/cmake/Config.cmake.in
+@@ -9,6 +9,10 @@ if (@UUID_SYSTEM_GENERATOR@)
+     endif ()
+ endif ()
+ 
++if (NOT "@UUID_USING_CXX20_SPAN@")
++    find_dependency(Microsoft.GSL)
++endif ()
++
+ include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake")
+ 
+ check_required_components(@PROJECT_NAME@)
+\ No newline at end of file

--- a/ports/stduuid/fix-install-directory.patch
+++ b/ports/stduuid/fix-install-directory.patch
@@ -1,0 +1,23 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index fffb92c..f45a7dc 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -11,7 +11,8 @@ option(UUID_USING_CXX20_SPAN "Using span from std instead of gsl" OFF)
+ add_library(${PROJECT_NAME} INTERFACE)
+ target_include_directories(${PROJECT_NAME} INTERFACE
+         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+-        $<INSTALL_INTERFACE:include>)
++        $<INSTALL_INTERFACE:include>
++        $<INSTALL_INTERFACE:include/stduuid>)
+ 
+ # Using system uuid generator
+ if (UUID_SYSTEM_GENERATOR)
+@@ -44,7 +45,7 @@ if (NOT UUID_USING_CXX20_SPAN)
+ endif ()
+ 
+ # Install step and imported target
+-install(FILES include/uuid.h DESTINATION include)
++install(FILES include/uuid.h DESTINATION include/stduuid)
+ install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}-targets)
+ install(EXPORT ${PROJECT_NAME}-targets
+         DESTINATION lib/cmake/${PROJECT_NAME})

--- a/ports/stduuid/fix-libuuid-dependency.patch
+++ b/ports/stduuid/fix-libuuid-dependency.patch
@@ -1,0 +1,32 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 039ee53..d4218a4 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -23,11 +23,8 @@ if (UUID_SYSTEM_GENERATOR)
+         find_library(CFLIB CoreFoundation REQUIRED)
+         target_link_libraries(${PROJECT_NAME} INTERFACE ${CFLIB})
+     else ()
+-        find_package(Libuuid REQUIRED)
+-        if (Libuuid_FOUND)
+-            target_include_directories(${PROJECT_NAME} INTERFACE ${Libuuid_INCLUDE_DIRS})
+-            target_link_libraries(${PROJECT_NAME} INTERFACE ${Libuuid_LIBRARIES})
+-        endif ()
++        find_package(unofficial-libuuid CONFIG REQUIRED)
++        target_link_libraries(${PROJECT_NAME} INTERFACE unofficial::UUID::uuid)
+     endif ()
+ endif ()
+ 
+diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
+index fb981d2..8a3c8ab 100644
+--- a/cmake/Config.cmake.in
++++ b/cmake/Config.cmake.in
+@@ -4,8 +4,7 @@ include(CMakeFindDependencyMacro)
+ if (@UUID_SYSTEM_GENERATOR@)
+     if (WIN32 OR APPLE)
+     else ()
+-        list (APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
+-        find_dependency(Libuuid REQUIRED)
++        find_dependency(unofficial-libuuid CONFIG)
+     endif ()
+ endif ()
+ 

--- a/ports/stduuid/portfile.cmake
+++ b/ports/stduuid/portfile.cmake
@@ -1,0 +1,40 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mariusbancila/stduuid
+    REF v1.2.2
+    SHA512 30970c25992e1ba35d96e3b2fc8530466c1070b8b913b8c37e9f698f39121a5a74361e2c4db4c2ba2feddb0ce9b2f14b78c4761cdac09b89a6a0117b179b08a7
+    HEAD_REF master
+    PATCHES
+        fix-install-directory.patch
+        fix-gsl-polyfill.patch
+        fix-libuuid-dependency.patch
+)
+
+# the debug build is not necessary, because stduuid deployed files are header-only
+set(VCPKG_BUILD_TYPE release)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        system-gen UUID_SYSTEM_GENERATOR
+    INVERTED_FEATURES
+        gsl-span UUID_USING_CXX20_SPAN
+)
+
+vcpkg_cmake_configure(SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS 
+        -DUUID_BUILD_TESTS=OFF
+        ${FEATURE_OPTIONS}
+)
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib)
+
+if("gsl-span" IN_LIST FEATURES)
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/stduuid/uuid.h" "#ifdef __cpp_lib_span" "#if 0")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/stduuid/uuid.h" "#include <span>" "#include <gsl/span>")
+else()
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/stduuid/uuid.h" "#ifdef __cpp_lib_span" "#if 1")
+endif()
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

--- a/ports/stduuid/vcpkg.json
+++ b/ports/stduuid/vcpkg.json
@@ -1,0 +1,33 @@
+{
+  "name": "stduuid",
+  "version": "1.2.2",
+  "description": "A C++17 cross-platform implementation for UUIDs",
+  "license": "MIT",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
+  "features": {
+    "gsl-span": {
+      "description": "Using span from gsl instead of std",
+      "dependencies": [
+        "ms-gsl"
+      ]
+    },
+    "system-gen": {
+      "description": "Enable operating system uuid generator",
+      "dependencies": [
+        {
+          "name": "libuuid",
+          "platform": "!osx & !windows"
+        }
+      ]
+    }
+  }
+}

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6824,6 +6824,10 @@
       "baseline": "2021-09-10",
       "port-version": 0
     },
+    "stduuid": {
+      "baseline": "1.2.2",
+      "port-version": 0
+    },
     "stftpitchshift": {
       "baseline": "1.4.1",
       "port-version": 0

--- a/versions/s-/stduuid.json
+++ b/versions/s-/stduuid.json
@@ -1,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "bd235773c953e5636a7b9f6356437daef6e5010c",
+      "version": "1.2.2",
+      "port-version": 0
+    }
+  ]
+}

